File: lib/models/report.dart
import 'package:uuid/uuid.dart';

class Report {
  final String reportId;
  final String convId;
  final String? anomalyReport;
  final dynamic anomalyTurn;
  final DateTime? created_at;
  final String? imageUrl;
  // 필요에 따라 필드 추가

  Report({
    required this.reportId,
    required this.convId,
    this.anomalyReport,
    this.anomalyTurn,
    this.created_at,
    this.imageUrl,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    final rawReport = json['anomalyReport'];
    String? safeReport;
    if (rawReport == null) {
      safeReport = null;
    } else if (rawReport is String) {
      safeReport = rawReport;
    } else {
      safeReport = rawReport.toString();
    }
    return Report(
      reportId: json['reportId'],
      convId: json['convId'],
      anomalyReport: safeReport,
      anomalyTurn: json['anomalyTurn'],
      created_at: json['created_at'] != null ? DateTime.parse(json['created_at']) : null,
      imageUrl: json['imageUrl'],
    );
  }
} 

File: fastapi-app/app/schemas/report.py
from pydantic import BaseModel, Field
from uuid import UUID
from typing import List, Optional, Any
from datetime import datetime

class AnomalyReportResponse(BaseModel):
    reportId: UUID
    convId: UUID
    anomalyReport: Optional[str] = Field(default=None, description="이상 보고서 내용")
    anomalyTurn: Optional[Any] = Field(default=None, description="이상이 발생한 대화 턴")
    created_at: Optional[datetime] = Field(default=None, description="리포트 생성 시간")
    imageUrl: Optional[str] = Field(default=None, description="대화가 발생한 이미지 URL")

class AnomalyReportListResponse(BaseModel):
    status: str = Field(default="ok", description="응답 상태")
    data: List[AnomalyReportResponse] = Field(description="이상 보고서 목록")